src/rotation.zig

```diff
@@ -507,13 +591,88 @@ pub const RotationModes = struct {
 		pub const id: []const u8 = "stairs";
 		var modelIndex: u16 = 0;
 
-		fn subBlockMask(x: u1, y: u1, z: u1) u8 {
+		inline fn subBlockMask(x: u1, y: u1, z: u1) u8 {
 			return @as(u8, 1) << ((@as(u3, x)*2 + @as(u3, y))*2 + z);
 		}
-		fn hasSubBlock(stairData: u8, x: u1, y: u1, z: u1) bool {
+		inline fn hasSubBlock(stairData: u8, x: u1, y: u1, z: u1) bool {
 			return stairData & subBlockMask(x, y, z) == 0;
 		}
 
+		fn rotateX(data: u16) u16 {
+			comptime var rotateTable: [8][3]u1 = undefined;
```

Comment:
My sincere hope here is that this table will be calculated once at compile time and used as lookup table at runtime, hence sparing us some math and sin cos lookups.

Reply:
Yes, this would probably be optimal.